spec/mailers/user_mailer_spec.rb

```diff
@@ -197,4 +197,53 @@
       end
     end
   end
+
+  describe "challenge assignment" do
+
+    let!(:gift_exchange) { create(:gift_exchange) }
+    let!(:collection) { create(:collection, challenge: gift_exchange, challenge_type: "GiftExchange") }
+    let!(:otheruser) { create(:user) }
+    let!(:offer) { create(:challenge_signup, collection: collection, pseud: otheruser.default_pseud) }
+    let!(:open_assignment) { create(:challenge_assignment, collection: collection, offer_signup: offer) }
+
+    let(:email) { UserMailer.challenge_assignment_notification(collection.id, otheruser.id, open_assignment.id).deliver }
+
+    # Test the headers
+    it 'should have a valid from line' do
```

Comment:
Not super important because the rest of this file seems to have it, but just an fyi: "should" isn't really necessary -- writing it like "has a valid from line" helps keeps things shorter.